### PR TITLE
Add globally modifiable feastol for PosyIneq

### PR DESCRIPTION
**Pull request by 1ozturkbe**
Pretty self-explanatory. I think it's useful to label the constraints that are supposed to be tight, and that in some cases (eg. MC simulations) it is nice to be able to control the tolerance of feasibility errors for constraints under substitutions.

### CODE REVIEW

**Comment by 1ozturkbe** [comment]
test models please

**Comment by 1ozturkbe** [comment]
test this please

**Comment by 1ozturkbe** [comment]
@bqpd I checked the tests locally, everything should be good now. Review and merge when ready.

**Comment by bqpd** [comment]
will review! in the meantime it's failing that test on cvxopt, maybe it needs a solvers-specific tolerance?

**Comment by 1ozturkbe** [comment]
yey finally. harder than I thought. merge when happy.

File: gpkit/constraints/tight.py [hunk 1]
```diff
@@ -11,12 +11,15 @@ class Tight(ConstraintSet):
...
-                 printwarning=False):
+                 printwarning=False, **kwargs):
         super(Tight, self).__init__(constraints)
         if reltol:
             self.reltol = reltol
         self.raiseerror = raiseerror
         self.printwarning = printwarning
+        if kwargs is not None:
+            for key, value in kwargs.items():
+                setattr(self, key, value)
```
Comment (bqpd) [inline gpkit/constraints/tight.py]: what kwargs are you thinking, here?

File: gpkit/nomials/math.py [hunk 2]
```diff
@@ -437,12 +445,11 @@ def _simplify_posy_ineq(self, hmap, pmap=None, allow_tautological=True):
...
             self.const_coeff = coeff  # pylint: disable=attribute-defined-outside-init
-        # TODO: 1e-6 is currently an arbitrary tolerance. It could follow
-        #       gp.check_result()'s default (1e-3) but that seeems quite lax!
-        if (allow_tautological and (coeff >= -1e-6 or np.isnan(coeff))
+        if (allow_tautological and
+                (coeff >= -1.*self.feastol or np.isnan(coeff))
                 and len(hmap) == 1):  # a tautological monomial!
             return None  # ValueError("tautological constraint: %s" % self)
-        elif coeff <= -1e-6:
+        elif coeff <= -1.*self.feastol:
```
Comment (bqpd) [inline gpkit/nomials/math.py]: I think I might change this to -self.feastol

**Comment by bqpd** [comment]
I'm ready to merge this (after it passes tests :crossed_fingers:), but @1ozturkbe could you check my changes are ok?

File: gpkit/nomials/math.py [hunk 3]
```diff
@@ -445,11 +445,10 @@ def _simplify_posy_ineq(self, hmap, pmap=None, allow_tautological=True):
...
             self.const_mmap = self.pmap.pop(const_idx)  # pylint: disable=attribute-defined-outside-init
             self.const_coeff = coeff  # pylint: disable=attribute-defined-outside-init
-        if (allow_tautological and
-                (coeff >= -1.*self.feastol or np.isnan(coeff))
+        if (allow_tautological and (coeff >= -self.feastol or np.isnan(coeff))
                 and len(hmap) == 1):  # a tautological monomial!
-            return None  # ValueError("tautological constraint: %s" % self)
-        elif coeff <= -1.*self.feastol:
+            return None  # was ValueError("tautological constraint: %s" % self)
+        elif coeff <= -self.feastol:
```
Comment (1ozturkbe) [inline gpkit/nomials/math.py]: It's funny, when I did that there were some errors, that's why I did -1*. But if this is passing tests good!

File: gpkit/constraints/tight.py [hunk 4]
```diff
@@ -17,9 +17,6 @@ def __init__(self, constraints, reltol=None, raiseerror=False,
             self.reltol = reltol
         self.raiseerror = raiseerror
         self.printwarning = printwarning
-        if kwargs is not None:
-            for key, value in kwargs.items():
-                setattr(self, key, value)
```
Comment (1ozturkbe) [inline gpkit/constraints/tight.py]: What do you mean @bqpd by 'parameters should be documented in interface'? I would like to be able to post-process my tight constraints by the names.

Comment (bqpd) [reply gpkit/constraints/tight.py]: that makes sense but I'm not sure how to connect it to this! could you give me an example? also we could probably merge this PR and bring that into another...

File: gpkit/nomials/math.py (hunk 3 above)
Comment (bqpd) [reply gpkit/nomials/math.py]: yeah I added a pylint exemption above this line!

File: gpkit/constraints/tight.py (hunk 4 above)
Comment (1ozturkbe) [reply gpkit/constraints/tight.py]: Sure I can make another PR for that. The idea is to comb through the untight constraints afterwards with the purpose of adding new constraints to tighten them, or to do some other post-processing (eg. using slacks to inform some other decision).

File: gpkit/nomials/math.py (hunk 3 above)
Comment (1ozturkbe) [reply gpkit/nomials/math.py]: Oh ok, I see now! Thanks, will keep that in mind in the future!

**Pull request merged**